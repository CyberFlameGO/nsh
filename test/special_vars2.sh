
echo "\$? ---------"
echo $?
echo bar | grep foo
echo $?
echo bar | grep bar
echo $?
echo "\$0 ---------"
echo $0
echo "\$1 ---------"
echo $1
echo "\$2 ---------"
echo $2

func() {
    echo "\$# ---------"
    echo $#
    echo "\$* ---------"
    for x in $*; do
        echo $x
    done
    echo "\$@ ---------"
    for x in $@; do
        echo $x
    done
    echo "\"\$*\" ---------"
    for x in "$*"; do
        echo $x
    done
    echo "\"\$@\" ---------"
    for x in "$@"; do
        echo $x
    done

    shift

    echo "\"\$@*\" ---------"
    for x in "$*"; do
        echo $x
    done
    echo "\"\$@\" ---------"
    for x in "$@"; do
        echo $x
    done

    echo "\$0 ---------"
    echo $0
    echo "\$1 ---------"
    echo $1
    echo "\$2 ---------"
    echo $2
}

func "spam egg" breakfast
