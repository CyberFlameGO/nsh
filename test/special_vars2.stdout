$? ---------
0
1
bar
0
$0 ---------
test/special_vars2.sh
$1 ---------

$2 ---------

$# ---------
2
$* ---------
spam
egg
breakfast
$@ ---------
spam
egg
breakfast
"$*" ---------
spam egg breakfast
"$@" ---------
spam egg
breakfast
"$@*" ---------
breakfast
"$@" ---------
breakfast
$0 ---------
test/special_vars2.sh
$1 ---------
breakfast
$2 ---------
